internal/dcache/metadata_manager/metadata_manager_impl.go

```diff
@@ -238,7 +239,12 @@ func (m *BlobMetadataManager) getFile(filePath string) (*dcache.FileMetadata, er
 		log.Debug("GetFile :: Failed to unmarshal JSON data: %v", err)
 		return nil, err
 	}
+
 	log.Debug("GetFile :: Successfully unmarshaled JSON data for file %s", path)
+	if metadata.State == dcache.Writing || metadata.State == dcache.Deleting {
+		return nil, errors.New(fmt.Sprintf("File is in %d state", metadata.State))
+	}
```

Comment:
metadata manager should not be checking the content of the file and take actions based on that.
See the todo @ L224, we will change it to return a []byte